cub/cub/device/dispatch/dispatch_streaming_reduce.cuh

```diff
@@ -0,0 +1,409 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+_CCCL_SUPPRESS_DEPRECATED_PUSH
+#include <cuda/std/functional>
+_CCCL_SUPPRESS_DEPRECATED_POP
+
+#include <cub/device/dispatch/dispatch_reduce.cuh>
+#include <cub/iterator/arg_index_input_iterator.cuh>
+#include <cub/iterator/constant_input_iterator.cuh>
+
+#include <thrust/iterator/iterator_adaptor.h>
+#include <thrust/iterator/tabulate_output_iterator.h>
+
+#include <cuda/std/type_traits>
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS // Do not document
```

Comment:
```suggestion
#ifndef _CCCL_DOXYGEN_INVOKED // Do not document
```
This changed some days ago.

Reply:
Good catch. Thanks for pointing that out! 👍